models/image.py

```diff
@@ -1,10 +1,133 @@
 
-from models.base import Base
+import sqlalchemy as sa
+from sqlalchemy import orm
+from sqlalchemy.dialects.postgresql import JSONB
 
+from models.base import Base, FileOnDiskMixin, SpatiallyIndexed
+
+
+class Image(Base, FileOnDiskMixin, SpatiallyIndexed):
 
-class Image(Base):
     __tablename__ = 'images'
 
+    exposure_id = sa.Column(
+        sa.ForeignKey('exposures.id'),
+        nullable=True,
+        index=True,
+        doc=(
+            "ID of the exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    exposure = orm.relationship(
+        'Exposure',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    source_images = orm.relationship(
+        'Image',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Images used to produce a multi-image object "
+            "(e.g., an images stack, reference, difference, super-flat, etc)."
+        )
+    )
```

Comment:
Doesn't this orm.relationship create a 1:1 as written?  It needs to be a 1:many.

Reply:
I need to add a many-to-many table.